lib/elixir/test/elixir/regex_test.exs

```diff
@@ -247,6 +247,9 @@ defmodule RegexTest do
     assert matches_escaped?("# lol")
 
     assert matches_escaped?("\\A.^$*+?()[{\\| \t\n\x20\\z #hello\u202F\u205F")
+
+    assert matches_escaped?("{}")
+    assert matches_escaped?("[]")
```

Comment:
We probably need more elaborate input to actually test `[]` escaping since this test passes on v1.2 with no patch.